### ISSUE TITLE
The GAP test file tst/testinstall/semicong.tst runs forever

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-packages/Semigroups#191]
Line 42 of the GAP test file `tst/testinstall/semicong.tst` runs forever in `unstable-3.0`. This is because it tries the method:

```
InstallMethod(Size,
Concatenation("for a ", _record.info_string, "congruence class"),
[_IsXCongruenceClass],
function(class)
  local p, tab;
  if not IsFinite(Parent(class)) then
    TryNextMethod();
  fi;
  p := Position(GenericSemigroupData(Parent(class)), Representative(class));
  tab := AsLookupTable(EquivalenceClassRelation(class));
  return Number(tab, n -> n = tab[p]);
end);
```

in `congpairs.gi` and the `IsFinite(Parent(class))` runs forever. The semigroup where this congruence class is defined is infinite, but does not know that it is finite. So, `IsFinite` is trying to enumerate the semigroup, which will never finish.

The library method finds the size of the congruence class in a different way (and instantaneously).

**Comment by james-d-mitchell** [comment]
Just to say that this is still an issue with the newer Todd-Coxeter code.

**Comment by wilfwilson** [comment]
Some discussion about the details of the test which currently fails, for anyone interested.

The semigroup created in this test is the fp semigroup `S`:
```
S = <x, y, z | xy = yx, xz = zx, yz = zy  # commutativity relations
                x^2 = x, y^3 = y,         # x and y are torsion
                xz = x, yz = y>           # z is an identity for x and y
```
The first two lines of relations give us that `S` is first constructed as a direct product of the trivial group, the cyclic group of order 2, and the natural numbers, and then the final line says it's a quotient of that direct product by those '`z` is an identity for `x` and `y`' relations.

Since no power of `z` appears as the left/right-hand side of a relation, it follows that `z` has infinite order in `S`, and so `S` is infinite.

A congruence `c` on `S` is created by the single generating pair `[x, y]`.

Lines 41-43:
```
gap> ec := EquivalenceClassOfElement(c, z);;
gap> Size(ec);
1
```
The generating pair is `[x, y]`. Thus we can only have something equivalent to `z` if we can multiply either `x` or `y` by something to reach `z`. But by looking at the relations of `S`, `z` never appears on its own as one side of a relation. Thus it is only equivalent to itself.

Lines 44-48:
```
gap> ec := EquivalenceClassOfElement(c, y);;
gap> z in ec;
false
gap> Size(ec);
5
```
Mathematically, we know that `z in ec` if and only if `y in EquivalenceClassOfElement(c, z)`, but we've determined that `EquivalenceClassOfElement(c, z)` contains only `z`. So `false` is correct. How can we determine it computationally? One way would be to enumerate `ec`. We know it contains at least `x` and `y` (thanks to the generating pair), and apparently there are five things in total.

**Comment by wilfwilson** [comment]
@mtorpey I was wondering – have you now solved this problem? (modulo actually integrating it into the package).

**Comment by mtorpey** [comment]
This is resolved by PR #258, and both the failing tests from `semicong.tst` are now included in `SemigroupsTestInstall`.

**Comment by wilfwilson** [comment]
Brilliant.
